### PR TITLE
feat(3000): Welcome panel

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

Some viewports can end up with really small banners which get very cramped.
Also we wanted a welcome banner. Merged into the same PR as I was working with both together at first, before deciding against the banner in the welcome page

## Changes

**For reviewers**: You can access it always simply with http://localhost:8000/home#panel=welcome

* Use a container query to change the layout optimising for space.
* Added a Welcome panel with its own flag that we can setup for existing posthog users only.
* Not married to the copy or list - really happy to have suggestions there.
* Adds stories for the side panel

 Screenshot 2023-11-29 at 16 23 14

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/layout/navigation-3000/sidepanel/panels/SidePanelWelcome.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    const { closeSidePanel } = useActions(sidePanelStateLogic)
+
+    return (
+        <div className="flex flex-col items-center justify-center m-4 my-10 flex-1">
+            <h1 className="font-bold text-4xl text-center">
+                Welcome to <span className="text-primary-3000">PostHog&nbsp;3000</span>
+            </h1>
+
+            <p className="max-w-120 text-center">
+                <b>PostHog 3000</b> is the codename for our revamped user experience. It's currently in beta, and we'd
```
Comment (Twixes) [inline frontend/src/layout/navigation-3000/sidepanel/panels/SidePanelWelcome.tsx:18]: I don't _think_ we'll consider it beta once we actually roll it out (considering there'll be no opt-out)

File: frontend/src/layout/navigation-3000/sidepanel/panels/SidePanelWelcome.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                    <IconInfo className="text-3xl" />
+                    <div className="flex-1">
+                        <h3 className="mb-0 font-semibold">In-app Docs</h3>
+                        <span>Get contextual information from our docs without needing to leave the app.</span>
+                    </div>
+                </li>
+                <li className="bg-bg-3000 border rounded p-4 flex gap-4 items-center">
+                    <IconSearch className="text-3xl" />
+                    <div className="flex-1">
+                        <h3 className="mb-0 font-semibold">Global Search</h3>
```
Comment (Twixes) [inline frontend/src/layout/navigation-3000/sidepanel/panels/SidePanelWelcome.tsx:50]: These titles should also be sentence case

Comment (Twixes) [inline frontend/src/layout/navigation-3000/sidepanel/panels/SidePanelWelcome.tsx:50]: Search + Commands (although both of these are _relaunched_ and not new features)

File: frontend/__snapshots__/lemon-ui-lemon-banner--error.png
Comment (Twixes) [inline frontend/__snapshots__/lemon-ui-lemon-banner--error.png:1]: The icon went poof, not sure if intended

File: frontend/src/layout/navigation-3000/sidepanel/sidePanelLogic.tsx [hunk 3]
```diff
@@ -38,6 +64,7 @@ export const sidePanelLogic = kea<sidePanelLogicType>([
 
                 tabs.push(SidePanelTab.Docs)
                 tabs.push(SidePanelTab.Settings)
+                tabs.push(SidePanelTab.Welcome)
                 tabs.push(SidePanelTab.Activation)
```
Comment (Twixes) [inline frontend/src/layout/navigation-3000/sidepanel/sidePanelLogic.tsx:68]: Welcome is the most ephemeral tab, hence should be very last

File: frontend/src/layout/navigation-3000/sidepanel/panels/SidePanelWelcome.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+                            mode.
+                        </span>
+                    </div>
+                </li>
+
+                <li className="bg-bg-3000 border rounded p-4 flex gap-4 items-center">
+                    <IconNotebook className="text-3xl" />
+                    <div className="flex-1">
+                        <h3 className="mb-0 font-semibold">Notebooks</h3>
+                        <span>Gather your adhoc queries, ideas, replays, and more into a single place.</span>
```
Comment (Twixes) [inline frontend/src/layout/navigation-3000/sidepanel/panels/SidePanelWelcome.tsx:37]: *ad-hoc
Broadly I don't think notebooks are as much about gathering things into a single place, but rather about exploring ideas and building conclusions

File: frontend/src/layout/navigation-3000/sidepanel/panels/SidePanelWelcome.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        <div className="flex flex-col items-center justify-center m-4 my-10 flex-1">
+            <h1 className="font-bold text-4xl text-center">
+                Welcome to <span className="text-primary-3000">PostHog&nbsp;3000</span>
+            </h1>
+
+            <p className="max-w-120 text-center">
+                <b>PostHog 3000</b> is the codename for our revamped user experience. It's currently in beta, and we'd
+                love to hear your feedback!
+            </p>
+            <ul className="space-y-4">
```
Comment (Twixes) [inline frontend/src/layout/navigation-3000/sidepanel/panels/SidePanelWelcome.tsx:21]: One other major thing is "🔵 New Nav/Top Bar"

**Review by Twixes — COMMENTED** [review]
This is very cool. As for the copy and design, gonna give @corywatilo a chance to poke at this

File: frontend/src/layout/navigation-3000/sidepanel/sidePanelLogic.tsx [hunk 6]
```diff
@@ -20,9 +20,35 @@ export const sidePanelLogic = kea<sidePanelLogicType>([
             activationLogic,
             ['isReady', 'hasCompletedAllTasks'],
         ],
+        actions: [sidePanelStateLogic, ['closeSidePanel']],
     }),
 
+    reducers(() => ({
+        welcomeAnnouncementAcknowledged: [
+            false,
+            { persist: true },
```
[the remark this answers is not in the document]
Comment (benjackwhite) [reply frontend/src/layout/navigation-3000/sidepanel/sidePanelLogic.tsx:29]: Yeah there is a todo about making the feature flag be linked to some sort of event property. 

The local persistence is just to account for ingestion delays

File: frontend/__snapshots__/lemon-ui-lemon-banner--error.png
Comment (benjackwhite) [reply frontend/__snapshots__/lemon-ui-lemon-banner--error.png:1]: Yeah its due to the changed layout, it sort of flexes down in the screenshots to be as small as possible and the smaller layout triggers the resize which removes the icon... very much so a storybook issue, not something that happens in reality that i could see

File: frontend/src/layout/navigation-3000/sidepanel/panels/SidePanelWelcome.tsx (hunk 5 above)
Comment (benjackwhite) [reply frontend/src/layout/navigation-3000/sidepanel/panels/SidePanelWelcome.tsx:21]: I guess it is 🤔 but then also not really :D I think that falls under the "general UI refresh". But I also think Cory is going to change all my words so waiting on his thoughts :D

**Comment by benjackwhite** [comment]
Update the copy based on the blog post and linked the post. Still super happy for suggestions (please go for an actual suggestion so its easy to just commit if we like it)

File: frontend/src/layout/navigation-3000/sidepanel/panels/SidePanelWelcome.tsx [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+                        <span>An updated toolbar with a modern design and dark mode compatibility</span>
+                    </div>
+                </li>
+                <li className="bg-bg-3000 border rounded p-4 flex gap-4 items-center">
+                    <IconSearch className="text-3xl" />
+                    <div className="flex-1">
+                        <h3 className="mb-0 font-semibold">Search + commands bar</h3>
+                        <span>
+                            A unified feature enabling navigation to most of PostHog's built-in features using keyboard
+                            shortcuts - try it with <KeyboardShortcut shift k />
```
Comment (thmsobrmlr) [inline frontend/src/layout/navigation-3000/sidepanel/panels/SidePanelWelcome.tsx:89]:
```suggestion
                            shortcuts - try it with <KeyboardShortcut shift k /> for the search and <KeyboardShortcut cmd shift k /> for the command bar
```

File: frontend/src/layout/navigation-3000/sidepanel/panels/SidePanelWelcome.tsx [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+
+            <p className="max-w-120 text-center">
+                <b>PostHog 3000</b> is the codename for our revamped user experience. You can{' '}
+                <Link to={blogPostUrl}>read more about it here</Link> but the headline features you will notice are:
+            </p>
+            <ul className="space-y-4">
+                <li className="bg-bg-3000 border rounded p-4 flex gap-4 items-center">
+                    <IconNight className="text-3xl" />
+                    <div className="flex-1">
+                        <h3 className="mb-0 font-semibold">Colors, data density, dark mode</h3>
```
Comment (Twixes) [inline frontend/src/layout/navigation-3000/sidepanel/panels/SidePanelWelcome.tsx:40]: The default browser h3 font size feels a bit off in 3000… I would suggest setting `text-base` here, but actually can we just set a `font-size: 1rem` override under `.posthog-3000` in `global.scss`?

File: frontend/src/layout/navigation-3000/sidepanel/panels/SidePanelWelcome.tsx [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+import { useActions } from 'kea'
+import posthog from 'posthog-js'
+import { useEffect } from 'react'
+
+import { KeyboardShortcut } from '../../components/KeyboardShortcut'
+import { sidePanelStateLogic } from '../sidePanelStateLogic'
+
+const blogPostUrl = 'https://posthog.com/blog/why-redesign'
+
+export const SidePanelWelcome = (): JSX.Element => {
```
[the remark this answers is not in the document]
Comment (joethreepwood) [reply frontend/src/layout/navigation-3000/sidepanel/panels/SidePanelWelcome.tsx:12]: Can we change the first item to just 'Dark Mode' That seems the stand-out feature for users. 

I think we can also drop Toolbar 3000, as that doesn't have any new functionality and is already announced in the changelog. 

If needs be, Search + Ne Navigation can be combined.

**Comment by Twixes** [comment]
Took a stab at this per your suggestion @benjackwhite Also ended up rewriting much of the copy. Here's a comparison:

| Before | After |
|--------|--------|
| Screenshot 2023-11-29 at 16 23 36 | Screenshot 2023-11-29 at 16 23 14 |

**Comment by joethreepwood** [comment]
Can we change the Notebooks text to: 

'Explore ideas more easily' please!

**Comment by Twixes** [comment]
Your wish is your command @joethreepwood

**Comment by corywatilo** [comment]
FYI I changed the tab to say "What's new?". We can use it for specific updates when we want to in the future. (Different than changelog.)

## Todos

- [ ] Get button to open in a new window (currently opens in sidebar)
- [ ] Add in @lottiecoxon art (when it's ready) - not a blocker to merging

File: frontend/src/layout/navigation-3000/sidepanel/panels/SidePanelWelcome.tsx [hunk 10]
```diff
@@ -0,0 +1,250 @@
...
+                                src={isDarkModeOn ? featureSearchDark : featureSearch}
+                                alt="Improved search"
+                                width="100%"
+                                style={{ borderTopLeftRadius: '0.25rem' }}
+                            />
+                        </div>
+                    </Card>
+                    <Card className="flex flex-col">
+                        <div className="flex-1">
+                            <Title>Command bar</Title>
```
Comment (Twixes) [inline frontend/src/layout/navigation-3000/sidepanel/panels/SidePanelWelcome.tsx:208]: The "command palette" is the old name… the new iteration is called the "command bar"

File: frontend/src/layout/navigation-3000/sidepanel/sidePanelLogic.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (Twixes) [reply frontend/src/layout/navigation-3000/sidepanel/sidePanelLogic.tsx:42]: Is this done now @benjackwhite? Can't see in code but maybe I've missed it

**Review by Twixes — APPROVED** [review]
The new layout is great! I addressed the to-dos (except Lottie's art) and made some extra tweaks, here's what this looks like now:
Screenshot 2023-12-05 at 17 54 44
This should be good to merge. I'm not sure if good to release, because I couldn't see the logic to set a person property on side panel close, but that's not PR-blocking.

**Comment by joethreepwood** [comment]
Not to rain on everyone's parade, but I quite strongly that this new welcome panel isn't readable and needs to be rolled back. 

Screenshot 2023-12-06 at 10 12 23

This is what I'm seeing when I log in and I feel it's almost completely unreadable. There's so much text on the page, no discernable pattern for a grid to understand it across the left and right hand sides of the page, and the level of contrast between the images and the actual text make it hard to know which is which. 

Even after looking at it for a few minutes and trying to read through item-by-item, I still find it hard to follow and really difficult to look at. And that's when I'm prepared to see PostHog 3000 - actual users will be even more overwhelmed because they won't expect this.

We could try to address the hierarchy by making the welcome message boxes order and highlight better, removing images - but my direct feedback is that we'd be best off reverting to the previous version and laying this information out as simply as possible.
